providers/snowflake/pyproject.toml

```diff
@@ -65,6 +65,7 @@ dependencies = [
     "snowflake-connector-python>=3.7.1",
     "snowflake-sqlalchemy>=1.4.0",
     "snowflake-snowpark-python>=1.17.0;python_version<'3.12'",
+    "snowflake-snowpark-python>=1.27.0;python_version>='3.12' and python_version<'3.13'",
```

Comment:
This is not a good idea to limit it to <3.13 - unless we also specify version >= 3.13. There is a high chance that library supports Python 3.13 and we should not prevent it "proactively" - the Python 3.13 PR of mine seems to pass with snowpark tests after I modified few other deps - like pyarrow etc.. 

And there are a lot of libraries that do not update their trove classifiers specifying that they are now compatible (because this is more of a metadata hint that real statement). So unless we know that snowpark does not work with Python 3.13, we should not preemptively limit it.

Reply:
We also didn't have a version for 3.12
I checked and the library is not present in our 3.12 constraints

For 3.12 I know it wasn't supported before classifier was added https://github.com/snowflakedb/snowpark-python/pull/2852

Official docs also don't specify Python 3.13 as supported
https://docs.snowflake.com/en/developer-guide/snowpark/python/setup

Reply:
Also - our general approach is "optimistic" - when it comes to upper-binding,  precisely because we test and verify a lot of things in our CI and there we can find issues. So when we make airfow Python 3.13 compliant in my PR - it already contains all the limitaions that we need to put in order for our tests to pass - and for us, this is "decision point we have" - passes our tests -> works. So when we actually merge Py 3.13, the limits in main will be already "good" and by the time we  release Airfow 3.1, we will also release the updated providers, and it means that it will be "consistent" with Py 3.13.